11_notify/willshen8/pkg/puppy/lostpuppy_test.go

```diff
@@ -0,0 +1,71 @@
+package puppy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func runTest(t *testing.T, payload []byte, expectedBody string, expectedHTTPCode int) {
+	req, err := http.NewRequest("POST", "/api/lostpuppy/", bytes.NewBuffer(payload))
+	if err != nil {
```

Comment:
Since you are already using `stretchr/testify`, I would import `require` as well as `assert` and use that to verify that the tests are set up properly:
```
require.NoError(t, err)
```

it does a `t.Fatal()` under the covers, so the test will not continue, and you get a nice symmetry where test setup is verified with `require` (which aborts the test) and `assert` for verifying the results (which does not abort)`.

I think if you are not using `testify`, you can use the basic Go `testing.T` functions, but if you are using `testify`, you may as well use all its features and not mix it with basic Go testing.

Reply:
Thanks, good to know.